cloud: add a note for rename (#13762)
pingcap/docs: tidb-cloud/tidb-cloud-release-notes.md

This page lists the release notes of [TiDB Cloud](https://www.pingcap.com/tidb-cloud/) in 2023.

## June 2, 2023

**General changes**

- In our pursuit to simplify and clarify, we've updated our product names:

>     - "TiDB Serverless Tier" is now "TiDB Serverless"
>     - "TiDB Dedicated Tier" becomes "TiDB Dedicated"
>     - "TiDB On-Premise" has been renamed to "TiDB Self-hosted"

Author: ```suggestion
    - "TiDB Cloud Serverless Tier" is renamed to "TiDB Serverless".
    - "TiDB Cloud Dedicated Tier" is renamed to "TiDB Dedicated".
    - "TiDB On-Premises" is renamed to "TiDB Self-Hosted".
```

> - In our pursuit to simplify and clarify, we've updated our product names:
> 
>     - "TiDB Cloud Serverless Tier" is renamed to "TiDB Serverless".
>     - "TiDB Cloud Dedicated Tier" is renamed to "TiDB Dedicated".
>     - "TiDB On-Premises" is renamed to "TiDB Self-Hosted".
> 
>     Enjoy the same great performance under these refreshed names. Your experience, our priority.

Author: ```suggestion
- In our pursuit to simplify and clarify, we have updated the names of our products:

    - "TiDB Cloud Serverless Tier" is now called "TiDB Serverless".
    - "TiDB Cloud Dedicated Tier" is now called "TiDB Dedicated".
    - "TiDB On-Premises" is now called "TiDB Self-Hosted".

    Enjoy the same great performance under these refreshed names. Your experience is our priority.
```
